From 640e61152f62b084eb5739fc4483e58468b8890f Mon Sep 17 00:00:00 2001
From: Suman Anna <s-anna@ti.com>
Date: Tue, 11 Nov 2014 18:06:26 -0600
Subject: [PATCH 1399/1587] virtio: fix memory leak of virtio ida cache layers

The virtio core uses an static ida named virtio_index_ida for
assigning index numbers to virtio devices during registration.
The ida core may allocate some internal ida cache layers upon
any ida assignment, and all these layers are truely freed only
upon the ida destruction. The virtio_index_ida is not destroyed
at present, leading to a minor memory leak, if the virtio core
is built as a module and atleast one virtio device is registered
and unregistered.

Fix this by invoking ida_destroy() in the virtio core module
exit.

Signed-off-by: Suman Anna <s-anna@ti.com>
[zou: Original patch taken from
ti-sdk-am335x-evm-08.00.00.00-Linux-x86-Install.bin]
Signed-off-by: Cao Zou <cao.zou@windriver.com>
---
 drivers/virtio/virtio.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/virtio/virtio.c b/drivers/virtio/virtio.c
index fed0ce1..64eba4f 100644
--- a/drivers/virtio/virtio.c
+++ b/drivers/virtio/virtio.c
@@ -249,6 +249,7 @@ static int virtio_init(void)
 static void __exit virtio_exit(void)
 {
 	bus_unregister(&virtio_bus);
+	ida_destroy(&virtio_index_ida);
 }
 core_initcall(virtio_init);
 module_exit(virtio_exit);
-- 
1.7.5.4

